src/Read/Reader.cpp

```diff
@@ -19,9 +19,27 @@ NAMESPACE_AVI20_READ_BEGIN
 class BasicInfoParser : public ParserBase
 {
 public:
-   BasicInfoParser( IStream& stream ) : ParserBase( stream ) {}
+   BasicInfoParser( IStream& stream )
+      : ParserBase( stream )
+      , _MoviPos( 0ULL )
+   {
+
+   }
+
+   bool SkipChunk( const ChunkHeader& ch )
+   {
+      if ( ch.fcc == FCC( 'movi' ) )
+      {
+         _MoviPos = ch.StartDataPos();
+         return true;
+      }
+      return false;
+   }
+
```

Comment:
readability -- the function name `SkipChunk` seems like it should be `ShouldSkipChunk`, because you're *asking* if the chunk should be skipped, not performing the skip

in that case, it's weird that the function has a side-effect of updating `_MoviPos`

one quick idea: maybe the function should be called `OnGotChunk`, meaning that it's a notification to the class that we're processing a chunk, and it can save off data if desired. This function could have the convention that the return value means 'skip' or 'don't skip', or a pass-by-reference parameter like `bool& skipThisChunk` or `bool& processThisChunk` might make it more readable.

Reply:
Okay, I think latest commit pretty-much does what you are suggesting.

Reply:
Okay, I removed this commit.